src/app/cli/src/graphql.ml

```diff
@@ -89,37 +147,70 @@ module Make (Program : Coda_inputs.Main_intf) = struct
     open Types
 
     let sync_state =
-      io_field "sync_status" ~typ:sync_status
+      io_field "syncStatus" ~typ:sync_status
         ~args:Arg.[]
         ~resolve:(fun {ctx= coda; _} () ->
           Deferred.return
-            (Inputs.Incr_status.Observer.value @@ Program.sync_status coda)
+            (Coda_incremental.Status.Observer.value @@ Program.sync_status coda)
           >>| Result.map_error ~f:Error.to_string_hum )
 
     let commands = [sync_state]
   end
 
   module Subscriptions = struct
-    let to_pipe observer =
-      let reader, writer =
-        Strict_pipe.(create (Buffered (`Capacity 1, `Overflow Drop_head)))
-      in
-      Incr_status.Observer.on_update_exn observer ~f:(function
-        | Initialized value ->
-            Strict_pipe.Writer.write writer value
-        | Changed (_, value) ->
-            Strict_pipe.Writer.write writer value
-        | Invalidated ->
-            () ) ;
-      (Strict_pipe.Reader.to_linear_pipe reader).Linear_pipe.Reader.pipe
-
     let new_sync_update =
-      subscription_field "new_sync_update"
-        ~doc:"Subscripts on sync update from Coda" ~deprecated:NotDeprecated
+      subscription_field "newSyncUpdate"
+        ~doc:"Subscribes on sync update from Coda" ~deprecated:NotDeprecated
         ~typ:Types.sync_status
         ~args:Arg.[]
         ~resolve:(fun {ctx= coda; _} ->
-          Program.sync_status coda |> to_pipe |> Deferred.Result.return )
+          Program.sync_status coda |> Coda_incremental.Status.to_pipe
+          |> Deferred.Result.return )
+
+    let new_block =
+      subscription_field "newBlock"
+        ~doc:
+          "Subscribes on a new block created by a proposer with a public key \
+           KEY"
+        ~typ:(non_null Types.block)
+        ~args:Arg.[arg "publicKey" ~typ:(non_null string)]
+        ~resolve:(fun {ctx= coda; _} public_key ->
+          let public_key = Public_key.Compressed.of_base64_exn public_key in
+          (* Pipes that will alert a subscriber of any new blocks throughout the entire time the daemon is on *)
+          let global_new_block_reader, global_new_block_writer =
+            Pipe.create ()
+          in
+          let init, _ = Pipe.create () in
+          Broadcast_pipe.Reader.fold (Program.transition_frontier coda) ~init
```

Comment:
this took me a while to parse, can you add comments explaining whats going here?

Reply:
yes, I did that for my next PR.